src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -551,4 +635,52 @@ private void runWithRetries(ContextualRunnable runnable, TableBlob blob, String
       throw e;
     }
   }
+
+  private long getLatestSequenceNum(TableId tableId) throws InterruptedException, DeltaFailureException {
+    RetryPolicy<Object> retryPolicy = new RetryPolicy<>()
+      .withMaxDuration(Duration.of(context.getMaxRetrySeconds(), ChronoUnit.SECONDS))
+      // maxDuration must be greater than the delay, otherwise Failsafe complains
+      .withBackoff(Math.min(61, context.getMaxRetrySeconds()) - 1, 120, ChronoUnit.SECONDS)
+      .withMaxAttempts(Integer.MAX_VALUE)
+      .onFailedAttempt(failureContext -> {
+        Throwable t = failureContext.getLastFailure();
+        LOG.error("Failed to read maximum sequence number from {}.{}.", tableId.getDataset(), tableId.getTable(), t);
+      });
+
+    try {
+      return Failsafe.with(retryPolicy).get(() -> {
+        if (bigQuery.getTable(tableId) == null) {
+          return 0L;
+        }
+
+        String query = String.format("SELECT MAX(_sequence_num) FROM %s.%s", tableId.getDataset(), tableId.getTable());
+        QueryJobConfiguration jobConfig = QueryJobConfiguration.of(query);
+        JobId jobId = JobId.of(UUID.randomUUID().toString());
+        Job queryJob = bigQuery.create(JobInfo.newBuilder(jobConfig).setJobId(jobId).build());
```

Comment:
Just curious, we are creating a big query job with a random UUID as jobId here? Wandering why it is required to have jobId here?

Reply:
This is from the example in the docs at https://cloud.google.com/bigquery/docs/quickstarts/quickstart-client-libraries#running_the_query. I haven't tried without setting the ID to see if BQ generates one for you.